common:
    lid_topic:  "/lidar_top" #"/velodyne_points_revise"
    imu_topic:  "/quadrotor_simulator_so3_2/imu"
    points_topic: "/lidar_top" #"/velodyne_points"
    odom_topic: "//odometry_gt_"
    time_sync_en: true         # ONLY turn on when external time synchronization is really not possible

dyn_obj:
    dataset: 1 #0 for kitti, 1 for nuscenes, 2 for waymo
    buffer_delay: 0.1
    buffer_size: 500000
    depth_map_dur: 0.2
    max_depth_map_num: 10
    max_pixel_points: 50
    frame_dur: 0.05
    self_x_f: 2.2
    self_x_b: -1.2
    self_y_l: 0.7
    self_y_r: -0.7
    blind_dis: 0.2
    fov_up: 10.0
    fov_down: -30.0
    fov_left: 180.0
    fov_right: -180.0
    fov_cut: 0
    points_num_perframe: 50000
    
    depth_thr1: 0.15
    enter_min_thr1: 0.5
    enter_max_thr1: 0.15
    map_cons_depth_thr1: 0.3
    map_cons_hor_thr1: 0.01
    map_cons_ver_thr1: 0.03
    map_cons_hor_dis1: 0.15
    map_cons_ver_dis1: 0.15
    depth_cons_depth_thr1: 0.5
    depth_cons_depth_max_thr1: 1.0
    depth_cons_hor_thr1: 0.02
    depth_cons_ver_thr1: 0.01
    enlarge_z_thr1 : -2.5
    enlarge_angle: 2.0
    enlarge_depth: 3.0
    occluded_map_thr1: 1
    case1_interp_en: true

    v_min_thr2: 1.0
    acc_thr2: 7.0
    map_cons_depth_thr2: 0.2
    map_cons_hor_thr2: 0.01
    map_cons_ver_thr2: 0.03
    occ_depth_thr2: 10.15
    occ_hor_thr2: 0.01
    occ_ver_thr2: 0.04
    depth_cons_depth_thr2: 0.1
    depth_cons_depth_max_thr2: 0.5
    depth_cons_hor_thr2: 0.01
    depth_cons_ver_thr2: 0.03
    occluded_times_thr2: 2
    k_depth2: 0.005
    case2_interp_en: false

    v_min_thr3: 0.5
    acc_thr3: 15.0
    map_cons_depth_thr3: 0.2
    map_cons_hor_thr3: 0.01
    map_cons_ver_thr3: 0.03
    occ_depth_thr3: 0.15
    occ_hor_thr3: 0.01
    occ_ver_thr3: 0.04
    depth_cons_depth_thr3: 0.3
    depth_cons_depth_max_thr3: 1.0
    depth_cons_hor_thr3: 0.01
    depth_cons_ver_thr3: 0.03
    occluded_times_thr3: 2
    k_depth3: 0.005
    case3_interp_en: false

    interp_hor_thr: 0.015
    interp_ver_thr: 0.06
    interp_thr1: 1.5
    interp_static_max: 10.0
    interp_start_depth1: 15
    interp_kp1: 0.15
    interp_kd1: 1.5
    interp_bg: 0.8
    interp_thr2: 0.25
    interp_thr3: 0.05

    point_index: -1 #16982 41299 #  #29614 #63177
    debug_x: 7.2065
    debug_y: -3.1844
    debug_z: -0.89005
    cluster_coupled: true
    cluster_future: true
    cluster_extend_pixel: 3
    cluster_min_pixel_number: 3

    # pose_file: "/home/huajie/event_detection/LiDAR-MOS/data/sequences/10/poses.txt"
    # out_file: "/home/huajie/event_detection/nuscenes/dataset/nuscenes5/09/predictions/"
    # out_file_origin: "/home/huajie/event_detection/nuscenes/dataset/nuscenes5_nc/09/predictions/"
    # # pc_file: "/home/huajie/event_detection/nuscenes/dataset/sequences/09/velodyne/"
    # label_file: "/home/huajie/event_detection/nuscenes/dataset/sequences/00/labels/"
    # # pred_file: "/home/huajie/event_detection/LiDAR-MOS/data/predictions_salsanext_1/sequences/10/predictions/"labels
    # pred_file: "/home/huajie/event_detection/nuscenes/dataset/nuscenes5/00/predictions/"
    # # vel_file: "/home/huajie/event_detection/nuscenes/dataset/sequences/00/vels/"

    # pose_log: true
    # pose_log_file: "/home/huajie/event_detection/nuscenes/dataset/sequences/09/pose.txt"

    # pose_file: "/media/yihang/LYH/kitti/dataset/sequences/00/poses.txt"0519
    # out_file: "/media/yihang/LYH/kitti/dataset/sequences/00/0507/"bag/revise/sequences/20/predictions/
    # pc_file : "/media/yihang/LYH/kitti/data_odometry_velodyne/dataset/sequences/00/velodyne/"
    # label_file : "/media/yihang/LYH/kitti/dataset/sequences/00/labels/"
    # pred_file : "/media/yihang/LYH/kitti/dataset/sequences/00/0507/"


    mode: 0 # 0 for distance resolution and 1 for angle resolution
    dyn_filter_en: false
    debug_publish: false
    
    interp_en: false
    is_add: false
    dyn_windows_num: 10
    dyn_windows_dur: 0.2
    checkneighbor_range: 3
    
    quadtree_layer_max: 4 ###
    ver_resolution_max: 0.022 #0.022
    hor_resolution_max: 0.007 #0.005
    angle_noise: 0.02 #couple with occu times
    
    min_dis_roll1: 0.3
    min_dis_pitch1: 0.1
    min_roll1: 0.02
    min_pitch1: 0.012
    depth_diff: 0.2
    depth_cut: 10.0
    roll_noise: 0.01
    pitch_noise: 0.005
    min_dis_roll2: 0.2
    min_dis_pitch2: 0.05
    min_roll2: 0.01
    min_pitch2: 0.005
    depth2: 0.5
    v_min_case2: 0.5
    depth_thr_case2: 0.15
    depth_minus: 0.5
    
    angle_occlude: 0.02
    depth_thr: 0.15
    depth_thr_fpc_max: 0.5
    depth_thr_fpc_min: 0.1


    
preprocess:
    lidar_type: 4                # 1 for Livox serials LiDAR, 2 for Velodyne LiDAR, 3 for ouster LiDAR, 4 for Realsense L515
    feature_extract_en: false
    scan_line: 6
    blind: 0

voxel:
    voxel_map_en: false
    pub_plane_en: false
    match_eigen_value: 0.005
    min_eigen_value: 0.005
    match_s: 0.9
    sigma_num: 81
    voxel_size: 3
    layer: 2

mapping:
    init_vel_x: 0.0
    init_vel_y: 0.0
    init_vel_z: 0.0
    imu_en: false
    acc_cov: 0.5
    gyr_cov: 0.2
    b_acc_cov: 0.0001
    b_gyr_cov: 0.0001
    fov_degree:    180
    det_range:     450.0
    extrinsic_T: [ 0.0, 0.0, -0.0 ]
    extrinsic_R: [ 1, 0, 0,
                   0, 1, 0,
                   0, 0, 1]

publish:
    path_en:  false
    scan_publish_en:  true       # false: close all the point cloud output
    dense_publish_en: true       # false: low down the points number in a global-frame point clouds scan.
    scan_bodyframe_pub_en: false  # true: output the point cloud scans in IMU-body-frame

pcd_save:
    pcd_save_en: false
    interval: -1                 # how many LiDAR frames saved in each pcd file; 
                                 # -1 : all frames will be saved in ONE pcd file, may lead to memory crash when having too much frames.